language-server/src/dotty/tools/languageserver/DottyLanguageServer.scala

```diff
@@ -441,6 +432,38 @@ class DottyLanguageServer extends LanguageServer
     }.asJava
   }
 
+  override def implementation(params: TextDocumentPositionParams) = computeAsync { cancelToken =>
+    val uri = new URI(params.getTextDocument.getUri)
+    val driver = driverFor(uri)
+    implicit val ctx = driver.currentCtx
+
+    val pos = sourcePosition(driver, uri, params.getPosition)
+    val uriTrees = driver.openedTrees(uri)
+    val path = Interactive.pathTo(uriTrees, pos)
+
+    val definitions = Interactive.findDefinitions(path, driver)
+    val projectsToInspect = projectsSeeing(definitions)
+
+    val originalSymbol = Interactive.enclosingSourceSymbol(path)
+    val implementations = {
+      val perProjectInfo = projectsToInspect.toList.map { config =>
+        val remoteDriver = drivers(config)
+        val ctx = remoteDriver.currentCtx
+        val definition = Interactive.localize(originalSymbol, driver, remoteDriver)
+        (remoteDriver, ctx, definition)
+      }
+
+      perProjectInfo.par.flatMap { (remoteDriver, ctx, definition) =>
```

Comment:
Similar concern about shadowing ctx here that I mentioned in the multi-project PR